crypto/mem_sec.c

```diff
@@ -29,7 +29,6 @@
 # include <sys/mman.h>
 # if defined(OPENSSL_SYS_LINUX)
 #  include <sys/syscall.h>
-#  include <linux/mman.h>
```

Comment:
use `#ifndef __ANDROID__` here.

Reply:
I'd say that `#ifdef SYS_mlock2` would be more appropriate. I mean you kind of detect if additional header is needed to start with. And in case Android chooses to expose SYS_mlock2 and define `MLOCK_ONFAULT` elsewhere, we'll notice, and only *then* use `__ANDROID__`...

Reply:
Why must include <linux / mman.h>, the code does not use the definition of the macro in the file.

Reply:
Here already includ <sys/mman.h> ， can not includ <linux/mman.h>
The contents of sys do not change as the kernel changes.
The contents of linux may change as the kernel changes.

Reply:
> Why must include <linux / mman.h>

For MLOCK_ONFAULT. sys/mman.h is common standardized interface and linux/mman.h is Linux-specific extension.

Reply:
So, Only use sys/mman.h is ok. The applications should not the use of Linux-specific extension.

Reply:
No, on Linux we want to use the Linux extensions.  Wrap it in ifndef OPENSSL_SYS_LINUX for example.

Reply:
But Linux extensions may change as the kernel changes. Need to check the linux version.

Reply:
> Need to check the linux version.

We do check for ENOSYS for mlock2 and fall back to mlock. And the check is sufficient, because kernel won't reuse syscall ordinal numbers.

Reply:
> Wrap it in ifndef OPENSSL_SYS_LINUX

It's already inside such ifdef. ifdef SYS_mlock2, naturally after sys/syscall.h is more versatile...